src/backend/distributed/utils/query_utils.c

```diff
@@ -40,12 +44,14 @@ ExtractRangeTableList(Node *node, ExtractRangeTableWalkerContext *context)
 	{
 		RangeTblEntry *rangeTable = (RangeTblEntry *) node;
 
-		/* make sure that we are extracting only relation entries if walkerMode is set to EXTRACT_RELATION_ENTRIES*/
-		if (walkerMode == EXTRACT_ALL_ENTRIES || (walkerMode ==
-												  EXTRACT_RELATION_ENTRIES &&
-												  rangeTable->rtekind == RTE_RELATION &&
-												  rangeTable->relkind ==
-												  RELKIND_RELATION))
+		/*
+		 * Make sure that we are extracting only relation entries if walkerMode is set to
+		 * EXTRACT_RELATION_ENTRIES. We treat partitioned tables as relations thoughout
+		 * the code, so fetch them as well.
+		 */
```

Comment:
@marcocitus what do you think about this change overall? I think #3110 was a good first step, but clearly missed some pieces.

This seems to be the way to go for me, `ExtractRangeTableRelationWalker ` returns all the RTE_RELATIONs that Citus might be interested in. 

I need to change few callers, which expects only distributed tables but relying on this function.